<launch>

    <!-- give a defulat value to arg -->
    <arg name="ext_cam" default="True" />

    <!-- to launch openni2 launch file from here -->
    <group if="$(arg ext_cam)">
        <include file="$(find openni2_launch)/launch/openni2.launch">
        </include>
    </group>

    <arg name="marker_id"        default="26"/> <!-- old 26 -->
    <arg name="marker_size"      default="0.141"/>    <!-- in m -->
    <!-- TODO:: how to give to different marker sizes ??-->
    <arg name="marker_frame"              default="marker"/> <!-- camera_rgb_optical_frame -->
    <arg name="camera_frame"                 default="camera_rgb_optical_frame"/>  <!-- leave empty and the pose will be published wrt param parent_name -->
    <!-- camera_rgb_optical_frame -->

    <!--<node pkg="tf" type="static_transform_publisher" name="link1_broadcaster" args="-0.24 0 0 0 0 0 left_gripper_base marker 30" />-->

    <!--<node pkg="tf" type="static_transform_publisher" name="link2_broadcaster" args="0.323 -0.141 -1.261 1.579 -0.804 -1.573 left_gripper_base base 10" /> -->



    <node pkg="aml_calib" type="aml_camera_pose_estimator" name="aml_camera_pose_estimator" output="screen">

        <remap from="/rgb_camera_info" to="/camera/rgb/camera_info" />
        <remap from="/rgb_rect_image" to="/camera/rgb/image_rect_color"/>

        <param name="image_is_rectified" value="True"/>
        <param name="marker_size"        value="$(arg marker_size)"/>
        <param name="marker_id"          value="$(arg marker_id)"/>
        <param name="camera_frame"                 value="$(arg camera_frame)"/>
        <param name="marker_frame"                 value="$(arg marker_frame)" />
    </node>

</launch>

